### ISSUE TITLE
Odd looking lighting when using a 16-bit png normal map

### ISSUE THREAD

**Issue by marechal-p** [godotengine/godot#41157]
**Godot version:** 3.2.2 stable

**OS/device including version:** Windows 10 1909

**Issue description:**

When using a normal map on a simple cube/plane surface, lighting appears as if being cut in diagonal, see:

![image](https://user-images.githubusercontent.com/14182238/89744351-0f8d3880-da7a-11ea-89a6-e8438e3102d7.png)

Note that I am using an omnilight. I cannot explain why it is shaded like this.

**Steps to reproduce:**

None?

**Minimal reproduction project:**

[weird-normal.zip](https://github.com/godotengine/godot/files/5048625/weird-normal.zip)

**Comment by clayjohn** [comment]
I can't tell if this is a bug with your normalmap or with Godot. I haven't seen this happen before on any other scenes. I was able to fix it in yours by converting the SpatialMaterial to a ShaderMaterial and then replacing the texture NORMALMAP line with : 
```
NORMALMAP = (texture(texture_normal,base_uv).rgb) * 2.0 - 1.0;
```
To me, this seems like there is an issue with how you encoded your normalmap. It looks like it is encoded with values from 0.5-1.0 instead of 0.0 to 1.0. Is that possible?

**Comment by steb-vs** [comment]
Found an interesting way to fix this.

From the reproduction project assets, open the "TexturesCom_Tiles_Plain_512_normal.png" normal map in Paint and save it again as PNG (same format).

Steps to reproduce:

Right click on "TexturesCom_Tiles_Plain_512_normal.png" > Edit
File > Save as
Set format to PNG, and set file name to "TexturesCom_Tiles_Plain_512_normal_2.png"
Open the reproduction project in Godot
Select the mesh instance > click on material 0
In the normal map section, replace the texture by the one saved with Paint

The lighting is now fixed:

![Capture](https://user-images.githubusercontent.com/14201278/89762979-d7c1d780-daf1-11ea-91c5-4adff97e3a75.PNG)

I don't really know what to conclude from that except that we have visually the same result when opening the 2 textures with IrfanView (as shown below), but not the same result once imported and used in Godot.

![Capture2](https://user-images.githubusercontent.com/14201278/89763242-69c9e000-daf2-11ea-8e01-ff81235df4a2.PNG)

**Comment by lawnjelly** [comment]
I found something which may be a clue. I loaded it up in my build of 3.2.3 (dev) version and it worked fine straight off.

Then I loaded it in 3.2 stable and it worked fine. Then I deleted the project, unzipped again so it could reimport, and the problem occurred in 3.2 stable. Then I loaded in my 3.2.3 and it still occurred, until I reimported the texture.

There is an important difference in my build, I had to compile it with the `builtin_libpng` flag (for some reason to get it working, maybe I didn't have the right png libraries on my linux). So I suspect the builtin libpng library *is working* with this particular flavour of png file, and the external library used for compiling most builds *is not*.

This ties it with it working once you re-exported the png from Paint.
